### PR TITLE
Fix different docs issues

### PR DESCRIPTION
**Pull request by theogf**
Now that the previous PR docs has been merged I noticed a lot of issues. This PR should fix most of them

### CODE REVIEW

**Comment by st--** [comment]
Is this a good time to turn on the strict docs check ?

File: docs/src/api.md [hunk 1]
```diff
@@ -37,9 +37,9 @@ The rest of the links [`ExpLink`](@ref), [`LogisticLink`](@ref), etc.,
 are aliases for the corresponding wrapped functions in a `Link`.
 For example `ExpLink == Link{typeof(exp)}`.
 
-When passing a [`Link`](@ref) to an `AbstractLikelihood`, this link 
+When passing a [`Link`](@ref) to a likelihood object, this link 
 corresponds to the transformation `p=link(f)` while, as mentioned in the
-[`Constrained parameters`](@ref) section, the statistics literature usually use
+[Constrained parameters](@ref) section, the statistics literature usually use
```
Comment (st--) [inline docs/src/api.md:42]:
```suggestion
[Constrained parameters](@ref) section, the statistics literature usually uses
```

File: docs/src/index.md [hunk 2]
```diff
@@ -14,42 +20,42 @@ taking a `Real` or an `AbstractVector` as an input and returns a
 Most likelihoods, like the [`GaussianLikelihood`](@ref), only require one latent Gaussian process.
 Passing a `Real` will therefore return a [`UnivariateDistribution`](https://juliastats.org/Distributions.jl/latest/univariate/),
 and passing an `AbstractVector{<:Real}` will return a [multivariate product of distributions](https://juliastats.org/Distributions.jl/latest/multivariate/#Product-distributions).
-```@repl
+```@repl test-repl
 f = 2.0;
-GaussianLikelihood()(f) == Normal(2.0)
-fs = [2.0, 3.0, 1.5]
-GaussianLikelihood()(fs) == Product([Normal(2.0), Normal(3.0), Normal(1.5)])
```
Comment (st--) [inline docs/src/index.md:21]: why did this have to be changed ?

Comment (st--) [reply docs/src/index.md:21]: same for the other docstring below

Comment (theogf) [reply docs/src/index.md:21]: I wanted to try something where the output would be more interesting but it ended up not being the best

Comment (theogf) [reply docs/src/index.md:21]: My first controlled `git revert` :smiling_face_with_tear:

**Comment by theogf** [comment]
> Is this a good time to turn on the strict docs check ?

I would keep it for later, right now I would just like to fix the current behavior

**Review by st-- — APPROVED** [review]

**Pull request merged**